python/tests/test_highlevel.py

```diff
@@ -948,6 +948,73 @@ def test_no_mrca(self):
         assert math.isinf(tree.path_length(0, 1))
 
 
+class TestPath:
+    t = tskit.Tree.generate_balanced(9)
+    #         16
+    #    ┏━━━━┻━━━┓
+    #    ┃       15
+    #    ┃     ┏━━┻━┓
+    #   11     ┃   14
+    #  ┏━┻━┓   ┃  ┏━┻┓
+    #  9  10  12  ┃ 13
+    # ┏┻┓ ┏┻┓ ┏┻┓ ┃ ┏┻┓
+    # 0 1 2 3 4 5 6 7 8
+
+    def test_tmrca_leaf(self):
+        assert self.t.path(0, 16) == [9, 11, 16]
+        assert self.t.path(16, 0) == [11, 9, 0]
+        assert self.t.path(7, 16) == [13, 14, 15, 16]
+
+    def test_equal_len_path(self):
+        assert len(self.t.path(5, 16)) == self.t.path_length(5, 16)
+
+    def test_two_leaves(self):
+        assert self.t.path(0, 8) == [9, 11, 16, 15, 14, 13, 8]
```

Comment:
It's not obvious why the second argument is included in the list here and not the first - why is ``8`` on the tail but 0 not on the front? It seems natural that ``t.path(u, v)``  should be equal to ``reversed(t.path(v, u))``.

I guess this is because of the requirement that ``len(t.path(u, v))`` == ``t.path_length(u, v)``? I guess this is because path_length counts the edges whereas this ``path`` function counts the nodes?

Reply:
Yes, I tried to fill the requirement, but I was not sure how. So I though about how we want to use it for B2, where we want the list to include the last node but not the first one, because the probability computation needs to use the ancestry nodes, like here:

```python
path = []
u = tree.parent(u)
while u != tskit.NULL:
      path.append(u)
      u = tree.parent(u)
return path
```

But indeed for a more general purpose the best would be to have none of the first and last nodes, or both of them.